### DISCUSSION TITLE
What is the best to make plugin settings site specific?

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/cms#16000] [Q&A] [answered]
Let's say that i am creating integration with google analytics and i need setting which stores analytics ID that will be used. However, analytics ID should be different depending on which site is used.

What would be best way to allow plugin users to set it up?

**Comment by brandonkelly** [comment, marked as answer]
The _easiest_ way would be to have them set the setting value from `config/<plugin-handle.php>`, where they can set it to an array indexed by site handle. Then when you are fetching the value, run it through `craft\helpers\ConfigHelper::localizedValue()`, which will check if it’s an array, and if so, only return the value for the current site.

**Comment by piotrpog** [comment]
@brandonkelly 
How about using the control panel interface? By default, we have just setting object there - should i maybe set specific object properties to json with arrays indexed by site handles, just like in the settings file?

**Reply by brandonkelly** [reply]
If you want it to be CP-editable, I’d suggest saving the values in the project config:

```php
$oldValue = Craft::$app->projectConfig->get('my-plugin-handle.setting-key') ?? $defaultValue;

Craft::$app->projectConfig->set('my-plugin-handle.setting-key', $newValue);
```
